pkg/graph/resolvers/system_intake_actions.go

```diff
@@ -0,0 +1,93 @@
+	ctx context.Context,
+	store *storage.Store,
+	fetchUserInfo func(context.Context, string) (*models.UserInfo, error),
+	input model.SystemIntakeRequestEditsInput,
+) (*models.SystemIntake, error) {
+	intake, err := store.FetchSystemIntakeByID(ctx, input.SystemIntakeID)
+	if err != nil {
+		return nil, err
+	}
+	// Set the intake current step to the requested edits step
+	formStepMap := map[model.SystemIntakeFormStep]models.SystemIntakeStep{
+		model.SystemIntakeFormStepInitialRequestForm: models.SystemIntakeStepINITIALFORM,
+		model.SystemIntakeFormStepDraftBusinessCase:  models.SystemIntakeStepDRAFTBIZCASE,
+		model.SystemIntakeFormStepFinalBusinessCase:  models.SystemIntakeStepFINALBIZCASE,
+	}
+	intake.Step = formStepMap[input.IntakeFormStep]
+	var targetForm models.GovernanceRequestFeedbackTargetForm
+	// Set the state of the requested form step and set the targeted feedback step
+	switch input.IntakeFormStep {
+	case model.SystemIntakeFormStepInitialRequestForm:
+		intake.RequestFormState = models.SIRFSEditsRequested
+		targetForm = models.GovernanceRequestFeedbackTargetIntakeRequest
+	case model.SystemIntakeFormStepDraftBusinessCase:
+		intake.DraftBusinessCaseState = models.SIRFSEditsRequested
+		targetForm = models.GovernanceRequestFeedbackTargetDraftBusinessCase
+	case model.SystemIntakeFormStepFinalBusinessCase:
+		intake.FinalBusinessCaseState = models.SIRFSEditsRequested
+		targetForm = models.GovernanceRequestFeedbackTargetFinalBusinessCase
+	default:
+		return nil, &apperrors.BadRequestError{
+			Err: fmt.Errorf("Cannot request edits on %s", input.IntakeFormStep),
+		}
+	}
+	intake, err = store.UpdateSystemIntake(ctx, intake)
+	if err != nil {
+		return nil, err
+	}
+	adminTakingAction, err := fetchUserInfo(ctx, appcontext.Principal(ctx).ID())
+	if err != nil {
+		return nil, err
+	}
+	_, err = store.CreateAction(ctx, &models.Action{
+		ActionType:     models.ActionTypeREQUESTEDITS,
+		ActorName:      adminTakingAction.CommonName,
+		ActorEUAUserID: adminTakingAction.EuaUserID,
+		ActorEmail:     adminTakingAction.Email,
+		BusinessCaseID: intake.BusinessCaseID,
+		IntakeID:       &intake.ID,
+		Feedback:       null.StringFrom(input.EmailFeedback),
+	})
+	if err != nil {
+		return nil, err
+	}
+	govReqFeedback := new(models.GovernanceRequestFeedback)
+	govReqFeedback.IntakeID = intake.ID
+	govReqFeedback.CreatedBy = adminTakingAction.EuaUserID
+	govReqFeedback.SourceAction = models.GovernanceRequestFeedbackSourceActionRequestEdits
+	govReqFeedback.TargetForm = targetForm
+	govReqFeedback.Feedback = input.EmailFeedback
+	_, err = store.CreateGovernanceRequestFeedback(ctx, govReqFeedback)
+	if err != nil {
+		return nil, err
+	}
+	if input.AdminNotes != nil {
+		_, err = store.CreateSystemIntakeNote(ctx, &models.SystemIntakeNote{
+			SystemIntakeID: intake.ID,
+			AuthorEUAID:    adminTakingAction.EuaUserID,
+			AuthorName:     null.StringFrom(adminTakingAction.CommonName),
+			Content:        null.StringFrom(*input.AdminNotes),
```

Comment:
Nitpick: you could use `Content: null.StringFromPtr(input.AdminNotes)` here. It shouldn't be a semantic difference, since you've got the `input.AdminNotes != nil` check already, just want to make sure you're aware of that function.

Reply:
Oh that's a good catch, thanks.

Reply:
resolved in [c38f771](https://github.com/CMSgov/easi-app/pull/2129/commits/c38f7719b2cd5c17d9a5d3b6c1b2479fef146da5)